aurora/src/main/java/com/market/banica/aurora/mapper/RequestMapper.java

```diff
@@ -0,0 +1,171 @@
+import com.orderbook.InterestsRequest;
+import com.orderbook.InterestsResponse;
+import com.orderbook.ItemOrderBookRequest;
+import com.orderbook.ItemOrderBookResponse;
+import com.orderbook.OrderBookServiceGrpc;
+import io.grpc.ManagedChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.management.ServiceNotFoundException;
+import java.rmi.NoSuchObjectException;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Service
+public class RequestMapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RequestMapper.class);
+
+    private final ChannelManager channelManager;
+    private final StubManager stubManager;
+
+    public static final String SPLIT_SLASH_REGEX = "/+";
+    public static final String SPLIT_EQUALS_REGEX = "=";
+    public static final String NUMBER_CHECK_REGEX = "^\\d+$";
+
+    public static final String ORDERBOOK = "orderbook";
+    public static final String AURORA = "aurora";
+    public static final String MARKET = "market";
+    public static final String BAD_PUBLISHER_REQUEST = "Unknown Publisher";
+    public static final String IN_CANCEL_ITEM_SUBSCRIPTION = "Forwarding to orderbook - cancelItemSubscription.";
+    public static final String IN_ANNOUNCE_ITEM_INTEREST = "Forwarding to orderbook - announceItemInterest.";
+    public static final String IN_GET_ORDER_BOOK_ITEM_LAYERS = "Forwarding to orderbook - getOrderBookItemLayers.";
+    public static final String IN_AURORA_REQUEST = "Forwarding to aurora - request.";
+
+    public static final String SUBSCRIBE = "subscribe";
+
+    public static final Pattern NUMBER_CHECK_PATTERN =
+            Pattern.compile(NUMBER_CHECK_REGEX);
+
+    @Autowired
+    public RequestMapper(StubManager stubManager, ChannelManager channelManager) {
+        this.stubManager = stubManager;
+        this.channelManager = channelManager;
+    }
+
+    public Aurora.AuroraResponse renderRequest(Aurora.AuroraRequest incomingRequest) throws NoSuchObjectException, ServiceNotFoundException {
+        String destinationOfMessage = incomingRequest.getTopic().split("/")[0];
+        Optional<ManagedChannel> channelByKey = channelManager.getChannelByKey(destinationOfMessage);
+
+        if (!channelByKey.isPresent()) {
+            throw new NoSuchObjectException("Channel " + destinationOfMessage + " is not available at the moment.");
+        }
+
+        if (destinationOfMessage.contains(ORDERBOOK)) {
+            return this.renderOrderbookMapping(incomingRequest, channelByKey.get());
+        } else if (destinationOfMessage.contains(AURORA)) {
+            return renderAuroraMapping(incomingRequest, channelByKey.get());
+        } else if (destinationOfMessage.contains(MARKET)){
+            throw new ServiceNotFoundException("Unimplemented publisher!");
```

Comment:
why is it market publisher unimplemented

Reply:
Because at this moment market publishers only supports subscribe features (sending tick responses for concreate item.).

It's left for future updates for example when an market can accomplish an buy/sell feature.